app/jobs/create_new_fake_sandbox_data_job.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class CreateNewFakeSandboxDataJob < CronJob
+  self.cron_expression = "0 2 * * *"
+
+  def perform
+    return unless Rails.env.sandbox? || Rails.env.test?
+
+    provider_name = "Education Development Trust"
+
+    edt_ecf_lead_provider = LeadProvider.find_by(name: provider_name)
+    edt_npq_lead_provider = NPQLeadProvider.find_by(name: provider_name)
+
+    school = edt_ecf_lead_provider.schools.first
```

Comment:
If I understand this correctly, you just want a random `LeadProvider` school here, right? First should order by id, and schools use uuid. In case I misunderstood and order is important you probably want something like `self.implicit_order_column = "created_at"` in your `School` model.